### PR TITLE
Isolate gemini, update to Node 12

### PR DESCRIPTION
**Pull request by gnomeontherun**
This moves Gemini stuff into its own directory with Node 10 setup specifically to support Gemini. Then the rest of the project is updated for Node 12. This is hopefully to fix the build memory issues we've been seeing recently.

## PR Checklist

Please check if your PR fulfills the following requirements:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

* [ ] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [x] Build related changes
* [x] CI related changes
* [ ] Documentation content changes
* [ ] clarity.design website / infrastructure changes
* [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

* [ ] Yes
* [ ] No

### CODE REVIEW

**Comment by hippee-lee** [comment]
> Where did the golden file removeal changes come from?

**Comment by gnomeontherun** [comment]
> Where did the golden file removeal changes come from?

Bumping dependencies caused the golden file to order properties differently in cases of setter/getter properties. Don't believe any other changes occurred.

File: golden/clr-angular.d.ts [hunk 1]
```diff
@@ -94,14 +94,14 @@ export declare class ClrAlert implements OnInit, OnDestroy, AfterViewInit {
     get alertClass(): string;
     set alertIconShape(value: string);
     alertTexts: QueryList<ElementRef>;
-    get alertType(): string;
```
[the remark this answers is not in the document]
Comment (gnomeontherun) [reply golden/clr-angular.d.ts]: Not removals, just `get` now comes after `set`. The diff is just lines moving up or down.

File: package.json [hunk 2]
```diff
@@ -143,10 +137,10 @@
   "devDependencies": {
     "@angular-builders/custom-webpack": "8.1.0",
     "@angular-builders/dev-server": "7.3.1",
-    "@angular-devkit/build-angular": "~0.900.1",
+    "@angular-devkit/build-angular": "git+https://git@github.com/angular/angular-devkit-build-angular-builds.git",
```
[the remark this answers is not in the document]
Comment (gnomeontherun) [reply package.json]: I did this to test the latest build while waiting for this week's release. The latest commit already goes back to normal.

**Pull request merged**